### PR TITLE
fix: fix formactions template to consider arguments

### LINKED ISSUE #143: FormActions ignoring css_class argument
Using `FormActions` with providing a `css_class` argument has no effect.

## Usage

```py
FormActions(
    CancelButton('/'),
    Submit('submit', _('Save')),
    css_class='d-flex justify-content-end',
),
```

## Output

```html
<div class="mb-3">
  <div class=""> 
    button1
    button2
  </div>
</div>
```

Comparing the bootstrap4 and bootstrap5 template shows that the `css_class` part is completely missing. In addition, I am wondering whether the inner `<div>` could be omitted.

### PR DESCRIPTION
**Pull request by mschoettle**
Adds `id` and `css_class` to the `div`. Also adds `row` to the class if the form is  supposed to be horizontal.

I removed the inner divs since those seemed to be not used. Let me know if that is incorrect.

Also, let me know if I should add tests for this.

Fixes #143

### CODE REVIEW

**Comment by JanMalte** [comment]
@smithdc1 could this please be merged and a bugfix version be released soon?

**Comment by smithdc1** [comment]
Hi All,

I'd like to see a small test added as that seems to be missing at the moment. A minimal layout could be created with the output asserted something like this. 

https://github.com/django-crispy-forms/crispy-bootstrap5/blob/94f0ca5adbe528b7ea43ab8238e4adc849ae84a3/tests/test_layout_objects.py#L366

**Comment by smithdc1** [comment]
@JanMalte  -- do you have time to add a test for this?

**Comment by mschoettle** [comment]
Sorry for the delay. I agree that tests would be good so I finally added some.

File: crispy_bootstrap5/templates/bootstrap5/layout/formactions.html [hunk 1]
```diff
@@ -1,9 +1,6 @@
-<div{% if formactions.attrs %} {{ formactions.flat_attrs }}{% endif %} class="mb-3">
-    {% if label_class %}
-        <div class="aab {{ label_class }}"></div>
-    {% endif %}
-
-    <div class="{{ field_class }}">
```
Comment (smithdc1) [inline crispy_bootstrap5/templates/bootstrap5/layout/formactions.html:6]: I think we still need to add `field_class`.

File: crispy_bootstrap5/templates/bootstrap5/layout/formactions.html [hunk 2]
```diff
@@ -1,9 +1,6 @@
...
+<div
+    {% if formactions.flat_attrs %}{{ formactions.flat_attrs }}{% endif %} 
+    class="mb-3{% if 'form-horizontal' in form_class %} row{% endif %} {{ formactions.css_class|default:'' }}"
```
Comment (smithdc1) [inline crispy_bootstrap5/templates/bootstrap5/layout/formactions.html:3]:
```suggestion
    class="mb-3{% if 'form-horizontal' in form_class %} row{% endif %} {{ formactions.css_class|default:'' }} {{ field_class }}" 
```

File: tests/results/test_formactions.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div
+     test="formactions-test"
+    class="mb-3 formactions-test-class"
```
Comment (smithdc1) [inline tests/results/test_formactions.html:3]:
```suggestion
    class="mb-3 formactions-test-class field-class" 
```

File: tests/test_layout_objects.py [hunk 4]
```diff
@@ -565,3 +565,45 @@ def test_grouped_checkboxes_radios(self):
...
+        test_form = SampleForm()
+        test_form.helper = FormHelper()
+        test_form.helper.form_tag = False
+        test_form.helper.layout = Layout(
+            FormActions(
+                HTML("<b>test</b>"),
+            ),
+        )
+
+        assert 'class="mb-3 "' in render_crispy_form(test_form)
```
Comment (smithdc1) [inline tests/test_layout_objects.py:579]:
```suggestion
        assert 'class="mb-3  "' in render_crispy_form(test_form)
```

File: tests/test_layout_objects.py [hunk 5]
```diff
@@ -565,3 +565,45 @@ def test_grouped_checkboxes_radios(self):
...
+        test_form.helper.form_tag = False
+        test_form.helper.form_class = 'form-horizontal'
+        test_form.helper.layout = Layout(
+            FormActions(
+                HTML("<b>test</b>"),
+                css_class="formactions-test-class",
+            ),
+        )
+
+        expected_class = 'class="mb-3 row formactions-test-class"'
```
Comment (smithdc1) [inline tests/test_layout_objects.py:608]:
```suggestion
        expected_class = 'class="mb-3 row formactions-test-class "'
```

File: tests/test_layout_objects.py [hunk 6]
```diff
@@ -565,3 +565,45 @@ def test_grouped_checkboxes_radios(self):
...
+                HTML("<b>test</b>"),
+            ),
+        )
+
+        assert 'class="mb-3 "' in render_crispy_form(test_form)
+
+    def test_formactions_attrs(self):
+        test_form = SampleForm()
+        test_form.helper = FormHelper()
+        test_form.helper.form_tag = False
```
Comment (smithdc1) [inline tests/test_layout_objects.py:584]:
```suggestion
        test_form.helper.form_tag = False
        test_form.helper.field_class = "field-class"
```

**Review by smithdc1 — CHANGES_REQUESTED** [review]
Thanks for the quick response. 

Just one small comment about `field_class`. I think that needs adding.

File: crispy_bootstrap5/templates/bootstrap5/layout/formactions.html (hunk 1 above)
Comment (mschoettle) [reply crispy_bootstrap5/templates/bootstrap5/layout/formactions.html:6]: I applied all the suggestions but then thought after thought that `field_class` might not be appropriate here: Would `field_class` not be applied to each child field of `FormActions`?

Consider the following test:

```py
def test_formactions_field_class(self):
        test_form = SampleForm()
        test_form.helper = FormHelper()
        test_form.helper.form_tag = False
        test_form.helper.field_class = "field-class"
        test_form.helper.layout = Layout(
            FormActions(
                Field('email'),
            ),
        )

        print(render_crispy_form(test_form))
```

Output:

```html
<div
     
    class="mb-3  field-class" 
    > <div id="div_id_email" class="mb-3"> <label for="id_email" class="form-label requiredField">
                email<span class="asteriskField">*</span> </label> <div class="field-class"> <input type="text" name="email" maxlength="30" class="textinput textInput inputtext form-control" required id="id_email"> <div id="hint_id_email" class="form-text">Insert your email</div> </div> </div> </div>
```

As per `field.html` the `<input>` gets wrapped into a div with CSS class `field-class`.

**Review by mschoettle — COMMENTED** [review]
Side note: I wasn't able to run `pytest` after following the README. `manage.py` is missing. The only way I got it running was by adding it to the root of the project. If this is acceptable I can add it in a separate PR.

**Comment by mschoettle** [comment]
@smithdc1 Brought this PR up to date with main. Could you please re-review?

File: crispy_bootstrap5/templates/bootstrap5/layout/formactions.html (hunk 1 above)
Comment (smithdc1) [reply crispy_bootstrap5/templates/bootstrap5/layout/formactions.html:6]: Is `FormActions` used to wrap `Field`s? The crispy-form [docs](https://django-crispy-forms.readthedocs.io/en/latest/layouts.html) say... 

>FormActions: It wraps fields in a <div class="form-actions">. It is usually used to wrap form’s buttons. 

With an example form as per the docs. (please forgive the lack of style on the cancel button).

Main branch:

image

This patch:

image

This therefore looks like a visual regression to me.

**Review by smithdc1 — COMMENTED** [review]
Hi @mschoettle 

Thanks for coming back to this! 🎁 

I've reviewed again and have added some images of what this looks like. I'm afraid I still think we need to keep `field_class`. Appreciate your thoughts.

File: crispy_bootstrap5/templates/bootstrap5/layout/formactions.html (hunk 1 above)
Comment (mschoettle) [reply crispy_bootstrap5/templates/bootstrap5/layout/formactions.html:6]: Thanks! I reverted that change and also brought back the div for `label_class` to be consistent with the other template packs.

**Comment by mschoettle** [comment]
@smithdc1 I addressed the requested changes. Could you please look at this again?

**Comment by mschoettle** [comment]
@smithdc1 Any chance you could re-review this?

**Comment by smithdc1** [comment]
Thanks for the ping. This looks good to me

One final question -- do you think this needs an entry in the changelog?

**Comment by smithdc1** [comment]
Added release note in #187

**Pull request closed without merging**